__tests__/Auth0Client.test.ts

```diff
@@ -41,66 +90,19 @@ const fetchResponse = (ok, json) =>
     json: () => Promise.resolve(json)
   });
 
-const setup = (
-  config?: Partial<Auth0ClientOptions>,
-  claims?: Partial<IdToken>
-) => {
-  const auth0 = new Auth0Client(
-    Object.assign(
-      {
-        domain: 'auth0_domain',
-        client_id: 'auth0_client_id',
-        redirect_uri: 'my_callback_url'
-      },
-      config
-    )
-  );
-
-  mockVerify.mockReturnValue({
-    claims: Object.assign(
-      {
-        exp: Date.now() / 1000 + 86400
-      },
-      claims
-    ),
-    user: {
-      sub: 'me'
-    }
-  });
-
-  return auth0;
-};
-
-const loginWithRedirect: any = async (
-  auth0,
-  tokenSuccess = true,
-  tokenResponse = {},
-  code = 'my_code',
-  state = 'MTIz'
-) => {
-  await auth0.loginWithRedirect();
-  expect(mockWindow.location.assign).toHaveBeenCalled();
-  window.history.pushState({}, '', `/?code=${code}&state=${state}`);
-  mockFetch.mockResolvedValueOnce(
-    fetchResponse(
-      tokenSuccess,
-      Object.assign(
-        {
-          id_token: 'my_id_token',
-          refresh_token: 'my_refresh_token',
-          access_token: 'my_access_token',
-          expires_in: 86400
-        },
-        tokenResponse
-      )
-    )
-  );
-  await auth0.handleRedirectCallback();
-};
+const setup = setupFn(mockVerify);
+const loginWithRedirect = loginWithRedirectFn(
+  mockWindow,
+  mockFetch,
+  fetchResponse
+);
+const loginWithPopup = loginWithPopupFn(mockWindow, mockFetch, fetchResponse);
+const checkSession = checkSessionFn(mockFetch, fetchResponse);
```

Comment:
Moved these functions to a separate file. However, as all of these needed access to some mocks, I created factories instead.